api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -186,115 +183,97 @@
-    this.userService.setClusterRetryCount(0);
-    this.leonardoNotebooksClient.deleteCluster(projectName, clusterName);
+  public ResponseEntity<EmptyResponse> deleteCluster(String workspaceNamespace) {
+    String firecloudWorkspaceName = lookupWorkspace(workspaceNamespace).getFirecloudName();
+    workspaceService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
+
+    this.leonardoNotebooksClient.deleteCluster(
+        workspaceNamespace, clusterNameForUser(userProvider.get()));
     return ResponseEntity.ok(new EmptyResponse());
   }
 
   @Override
   public ResponseEntity<ClusterLocalizeResponse> localize(
-      String projectName, String clusterName, ClusterLocalizeRequest body) {
+      String workspaceNamespace, ClusterLocalizeRequest body) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
     workspaceService.enforceWorkspaceAccessLevel(
-        body.getWorkspaceNamespace(), body.getWorkspaceId(), WorkspaceAccessLevel.READER);
-    workspaceService.validateActiveBilling(body.getWorkspaceNamespace(), body.getWorkspaceId());
+        dbWorkspace.getWorkspaceNamespace(),
+        dbWorkspace.getFirecloudName(),
+        WorkspaceAccessLevel.WRITER);
+    workspaceService.validateActiveBilling(
+        dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName());
 
-    FirecloudWorkspace fcWorkspace;
+    FirecloudWorkspace firecloudWorkspace;
     try {
-      fcWorkspace =
+      firecloudWorkspace =
           fireCloudService
-              .getWorkspace(body.getWorkspaceNamespace(), body.getWorkspaceId())
+              .getWorkspace(dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName())
               .getWorkspace();
     } catch (NotFoundException e) {
       throw new NotFoundException(
           String.format(
               "workspace %s/%s not found or not accessible",
-              body.getWorkspaceNamespace(), body.getWorkspaceId()));
+              dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName()));
     }
-    DbCdrVersion cdrVersion =
-        workspaceService
-            .getRequired(body.getWorkspaceNamespace(), body.getWorkspaceId())
-            .getCdrVersion();
+    DbCdrVersion cdrVersion = dbWorkspace.getCdrVersion();
 
     // For the common case where the notebook cluster matches the workspace
     // namespace, simply name the directory as the workspace ID; else we
     // include the namespace in the directory name to avoid possible conflicts
     // in workspace IDs.
-    String gcsNotebooksDir = "gs://" + fcWorkspace.getBucketName() + "/notebooks";
-    Timestamp now = new Timestamp(clock.instant().toEpochMilli());
-    long workspaceId =
-        workspaceService
-            .getRequired(body.getWorkspaceNamespace(), body.getWorkspaceId())
-            .getWorkspaceId();
+    String gcsNotebooksDir = "gs://" + firecloudWorkspace.getBucketName() + "/notebooks";
```

Comment:
I'd make a helper method or service method for this bucket path determination. (If `FirecloudWorkspace` weren't generated, I'd put it there.)